src/server.js

```diff
@@ -1,16 +1,19 @@
 // @flow
 
+import fs from 'fs';
+import path from 'path';
+import pify from 'pify';
 import { renderToString } from 'react-dom/server';
 import Styletron from 'styletron-server';
 import { StyletronProvider } from 'styletron-react';
-import requireDir from 'require-dir';
-
 import doc from 'lib/__html';
 
-const pages = requireDir('./pages');
+const readFile = pify(fs.readFile);
 
-export default (state: { page: string }): string => {
-    const Page = pages[state.page].default;
+const renderPage = async function renderPage(pageName) {
+    const page = await import(`./pages/${pageName}`);
```

Comment:
@sndrs What do you think of code splitting here instead of requiring the `pages` directory?

Reply:
does `import` work like this in node?

Reply:
i went with the whole dir import because it would put all the modules into memory once this file was `required`. maybe that was a bit preemptive?

Reply:
Okay, that's a good point! I'll change it back

Reply:
If you ever want to do the importing client side then you’ll probably have to keep this as a dynamic import and use the Airbnb dynamic import plugins for webpack and node in the different environments respectively. If you’re only ever server rendering then using Node specific importing will be fine.